backend/lightning/lightning.go

```diff
@@ -203,3 +236,28 @@ func (lightning *Lightning) setLightningConfig(config config.LightningConfig) er
 
 	return nil
 }
+
+func fetchAuthData(httpClient *http.Client, endpoint string) (*[]uint8, error) {
```

Comment:
no reason to return a pointer, just return `[]uint8`. at call site you could add a sanity check that the result is not empty.

i feel like this function body (http request, limited size, StatusOK checkm ...) appears many times in the codebase. can we just have a helper function in `util/`?